docs/joss_paper/paper.md

```diff
@@ -0,0 +1,155 @@
+    Maison de la Simulation,
+    91191, Gif-sur-Yvette, France
+  index: 3
+- name: >-
+    Max-Planck-Institut für Plasmaphysik, Garching,
+    Germany
+  index: 4
+- name: CINES, France
+  index: 5
+
+date: 24 April 2025
+bibliography: paper.bib
+
+---
+
+## Summary
+
+Gyselalib++ provides the mathematical building blocks to construct gyrokinetic plasma simulations in C++, simulating a distribution function discretised in phase space on a fixed grid.
+It relies on the DDC library [@ddc] to statically type the discretisation dimensions; thus preventing many common sources of errors.
+Via DDC, Gyselalib++ also leverages the Kokkos framework [@trott2022], ensuring portability across both CPU and various GPU architectures.
+The library provides a variety of tools including semi-Lagrangian advection operators, solvers for partial differential equations (PDEs), quadrature rules, and time steppers.
+All operators are designed to function correctly or raise compiler errors on non-orthonormal coordinate systems thanks to the static typing.
+
+## Statement of Need
+
+Plasma simulations are essential for the development of magnetic confinement fusion devices for energy production.
+The low collisionality of such plasmas make kinetic models a judicious choice.
+In particular gyrokinetic theory [@brizard2007; @krommes2012], which reduces the 6D problem to a 5D problem by removing high frequency dynamics, is a popular tool for such simulations [@garbet2010].
+Despite the reduction in dimensionality such simulations still require massively powerful high-performance computing (HPC) resources.
+For ITER-sized simulations, exascale resources would still be required.
+
+The pre-existing Gysela code [@grandgirard2016], written in Fortran, originally aimed to simulate plasma in the core region of a tokamak using semi-Lagrangian advection with a distribution function discretised in phase space on a fixed grid.
+This approach was shown to work well and allowed the study of many interesting physical phenomena [@Esteve2018;@Sarazin2021;@DifPradalier2022].
+However expanding this code to use more complex mathematical methods such as non-uniform points (vital for handling the different magnitudes of physical quantities in the core and edge regions), and increasingly complex geometries (such as D-shape geometries, geometries including both closed and open field lines, and potentially stellarator geometries) has proved to be challenging and sometimes error-prone.
+The challenges of such extensions are further amplified when trying to organise such a code for use on new GPU architectures, necessary for exascale simulations.
+This is a challenge shared by other gyrokinetic codes [@trilaksono2023].
+
+In the case of Gysela, several of the required changes would have affected a large percentage of the code base.
+This means that the effort required was comparable to a complete rewrite, without the advantages such a rewrite can bring.
```

Comment:
This sentence is not very clear to me. Can you rephrase this?

Reply:
Is this clearer: 17d773cfe ?

Reply:
Better for me, thanks